app/services/katello/product_content_finder.rb

```diff
@@ -16,6 +16,7 @@ def initialize(params = {})
     def product_content
       if match_environment
         versions = consumable.content_view_environments.select(:content_view_version_id).map(&:content_view_version_id)
+        environment = consumable.respond_to?(:environment) ? consumable.environment : consumable.content_view_environments.select(:environment_id).map(&:environment_id)
```

Comment:
I am afraid this looks a little bit hacky. I would rather implement `environment()` on `SubscriptionFacet` as well so we do not have to differentiate, here.

Problem there is that I am unsure on if we can determine a single value for environment for SubscriptionFacet as it gets `content_version_environments` through `ContentFacet`, which is a `has_many` relation.

Reply:
What about putting that in a condition to only evaluate if structured apt is enabled? The risk of breaking something for other OSes would be lower.

Reply:
You could even move the line to line 29 such that it will only be in the structured apt block.

Reply:
Even if we move it into the structured APT block, we still need to consider the case where some of the content uses structured APT, and some of it does not. In that case we enter the block, which should then not influence how the non structured APT content is handled.

Reply:
This is exactly the reason to move the code into the structured apt block. If we have the structured apt code in one block, that will significantly simplify testing because the rest of the code is not affected.